test/test_torch.py

```diff
@@ -780,6 +780,20 @@ def test_remainder(self):
         long_res1 = long_m1.clone()
         long_res1.remainder_(long_qs.unsqueeze(0).expand_as(long_res1))
 
+    def test_cross_type_binary_operators(self):
+        x = torch.randn(2, 2, dtype=torch.float)
+        y = torch.tensor(2., dtype=torch.double)
+
+        operators = [
+            operator.add, operator.sub, operator.mul, operator.truediv,
+            operator.pow, operator.mod, operator.lt, operator.le, operator.gt,
+            operator.ge, operator.eq, operator.ne
+        ]
+
+        for arg1, arg2, arg_same_type in [(x, y, y.float()), (y, x, x.double())]:
```

Comment:
can we check all the unify type cases?  This is missing at least the integral/floating mismatch case.

Reply:
What do you mean by all the unify type cases? Loop over the full matrix of types?

Reply:
Sorry for not being clear; I meant all the case statements that are in unifyTypes; at a minimum you are missing:
```
if (isFloatingType(scalar_tensor.type().scalarType()) && isIntegralType(nonscalar_tensor.type().scalarType())) {
```
this might be the only one; I didn't study it that closely.

Reply:
Right, good point. I'll fix that once we resolve where the dispatch should be.